test/unit/stats.js

```diff
@@ -0,0 +1,891 @@
+        assert.equal(err.status, 400);
+        assert.lengthOf(data, statCount);
+        assert.deepEqual([fakeStat, fakeStat2, fakeStat3,
+          fakeStat4, fakeStat5, fakeStat6]);
+        done();
+      });
+    });
+
+    // check error is thrown if required field 'student_id' is missing in request
+    xit('attempting to post stats with missing field student_id will result in an error', function(done) {
+      var req = {
+        param: {
+          event_id: 1,
+        },
+        body: {
+          pacer: 5,
+          height: 5,
+          weight: 5,
+        }
+      };
+
+      var promise = stats.getStats({});
+      var statCount;
+
+      promise.then(function(data) {
+        statCount = data.length;
+        assert.deepEqual(data, [fakeStat, fakeStat2, fakeStat3,
+          fakeStat4, fakeStat5, fakeStat6]);
+
+        return stats.createStats(req);
+      })
+      .catch(function() {
+        return stats.getStats({});
+      })
+      .catch(function(data) {
+        assert.equal(err.message,
+        'Could not post due to missing fields');
+        assert.equal(err.status, 400);
+        assert.lengthOf(data, statCount);
+        assert.deepEqual([fakeStat, fakeStat2, fakeStat3,
+          fakeStat4, fakeStat5, fakeStat6]);
+        done();
+      });
+    });
+  });
+
+  describe('updateStat(req)', function() {
+    // update existing stats
+    xit('should update stats in the database', function(done) {
+      var req = {
+        param: {
+          stat_id: 1,
+        },
+        body: {
+          height: 6,
+          weight: 6,
+          pacer: 6,
+        }
+      };
+
+      var promise = stats.getStats({});
+      var oldDB;
+      var statCount;
+
+      promise.then(function(data) {
+        statCount = data.length;
+        oldDB = data;
+        assert.deepEqual(data, [fakeStat, fakeStat2, fakeStat3,
+          fakeStat4, fakeStat5, fakeStat6]);
+
+        return stats.updateStat(req);
+      })
+      .catch(function() {
+        return stats.getStats({});
+      })
+      .catch(function(data) {
+        assert.lengthOf(data, statCount);
+        assert.notDeepEqual(data, oldDB);
+        assert.deepEqual(data, [fakeStat8, fakeStat2, fakeStat3,
+          fakeStat4, fakeStat5, fakeStat6]);
+        done();
+      });
+    });
+
+    // update non existing stats should error
+    xit('should error updating non-existing stats', function(done) {
+      var req = {
+        param: {
+          stat_id: 99999,
+        },
+        body: {
+          height: 6,
+          weight: 6,
+          pacer: 6,
+        }
+      };
+
+      var promise = stats.getStats({});
+      var oldDB;
+      var statCount;
+
+      promise.then(function(data) {
+        statCount = data.length;
+        oldDB = data;
+        assert.deepEqual(data, [fakeStat, fakeStat2, fakeStat3,
+          fakeStat4, fakeStat5, fakeStat6]);
+
+        return stats.updateStat(req);
+      })
+      .catch(function() {
```

Comment:
Same pattern as below here: the first block after updateStat should be a .catch with an err parameter passed to function. Put the asserts about err in the catch block. Then call return stats.getStats({}) in the catch block. After that, the second block should be a .then (since getStats works) with a parameter of data passed to function() [the way you have it already]. The asserts about the data should stay where they are.